numpy/lib/npyio.py

```diff
@@ -728,41 +728,47 @@ def _savez(file, args, kwds, compress, allow_pickle=True, pickle_kwargs=None):
     zipf.close()
 
 
+def _floatconv(x):
+    try:
+        return float(x)  # The fastest path.
+    except ValueError:
+        if '0x' in x:  # Don't accidentally convert "a" ("0xa") to 10.
+            try:
+                return float.fromhex(x)
+            except ValueError:
+                pass
+        raise  # Raise the original exception, which makes more sense.
+
+
+def _boolconv(x): return bool(int(x))
+def _intconv(x): return int(float(x))
+def _complexconv(x): return complex(asstr(x).replace('+-', '-'))
+
+
+_CONVERTERS = [
+    (np.bool_, _boolconv),
+    (np.uint64, np.uint64),
+    (np.int64, np.int64),
+    (np.integer, _intconv),
+    (np.longdouble, np.longdouble),
+    (np.floating, _floatconv),
+    (complex, _complexconv),
+    (np.bytes_, asbytes),
+    (np.unicode_, asunicode),
+]
```

Comment:
At this point you could inline the lambdas into the table without causing any issues.

Reply:
Yup.